operator/internal/manifests/config.go

```diff
@@ -54,9 +60,33 @@ func ConfigOptions(opt Options) config.Options {
 		rwConfig                   *config.RemoteWriteConfig
 	)
 
-	if rulerEnabled {
+	if opt.Stack.Tenants != nil && opt.Ruler.OCPAlertManagerEnabled && opt.Stack.Tenants.Mode == lokiv1.OpenshiftLogging {
 		rulerEnabled = true
 
+		if opt.Ruler.Spec == nil {
+			opt.Ruler.Spec = &lokiv1beta1.RulerConfigSpec{
+				AlertManagerSpec: &lokiv1beta1.AlertManagerSpec{},
+			}
+		}
+
+		if opt.Ruler.Spec.AlertManagerSpec == nil || len(opt.Ruler.Spec.AlertManagerSpec.Endpoints) == 0 {
+			ams := &lokiv1beta1.AlertManagerSpec{
+				Endpoints: []string{"https://_web._tcp.alertmanager-operated.openshift-monitoring.svc"},
+				EnableV2:  true,
+				DiscoverySpec: &lokiv1beta1.AlertManagerDiscoverySpec{
+					EnableSRV:       true,
+					RefreshInterval: "1m",
+				},
+			}
+
+			if err := mergo.Merge(opt.Ruler.Spec.AlertManagerSpec, ams); err != nil {
+				return config.Options{}, kverrors.Wrap(err, "failed merging RulerSpec options")
+			}
+		}
+
+	}
```

Comment:
The general pattern to apply openshift partial patches for automation looks like this:
1. Create a configure method in the openshift package
2. Apply the configure after `ConfigOptions()` returns and handle any error

See ConfigureDeploymentForTenantMode.